tests/e2e/main.go

```diff
@@ -0,0 +1,380 @@
+		var runAsUser int64 = 65534
+
+		maoManifest := &appsv1beta2.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "machine-api-operator",
+				Namespace: "kube-system",
+				Labels: map[string]string{
+					"k8s-app":                     "machine-api-operator",
+					"managed-by-channel-operator": "true",
+				},
+			},
+			Spec: appsv1beta2.DeploymentSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"k8s-app": "machine-api-operator",
+					},
+				},
+				Template: apiv1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"k8s-app":                     "machine-api-operator",
+							"managed-by-channel-operator": "true",
+						},
+					},
+					Spec: apiv1.PodSpec{
+						Containers: []apiv1.Container{
+							{
+								Name:  "machine-api-operator",
+								Image: maoImage,
+								Command: []string{
+									"/machine-api-operator",
+								},
+								Args: []string{
+									"-v=4",
+								},
+								Resources: apiv1.ResourceRequirements{
+									Limits: apiv1.ResourceList{
+										"cpu":    resource.MustParse("20m"),
+										"memory": resource.MustParse("50Mi"),
+									},
+									Requests: apiv1.ResourceList{
+										"cpu":    resource.MustParse("20m"),
+										"memory": resource.MustParse("50Mi"),
+									},
+								},
+								VolumeMounts: []apiv1.VolumeMount{
+									{
+										Name:      "mao-config",
+										MountPath: "/etc/mao-config",
+									},
+								},
+							},
+						},
+						ImagePullSecrets: []apiv1.LocalObjectReference{
+							{
+								Name: "coreos-pull-secret",
+							},
+						},
+						NodeSelector: map[string]string{
+							"node-role.kubernetes.io/master": "",
+						},
+						RestartPolicy: "Always",
+						SecurityContext: &apiv1.PodSecurityContext{
+							RunAsNonRoot: &runAsNonRoot,
+							RunAsUser:    &runAsUser,
+						},
+						Tolerations: []apiv1.Toleration{
+							{
+								Key:      "node-role.kubernetes.io/master",
+								Operator: "Exists",
+								Effect:   "NoSchedule",
+							},
+						},
+						Volumes: []apiv1.Volume{
+							{
+								Name: "mao-config",
+								VolumeSource: apiv1.VolumeSource{
+									ConfigMap: &apiv1.ConfigMapVolumeSource{
+										LocalObjectReference: apiv1.LocalObjectReference{
+											Name: "mao-config-v1",
+										},
+										Items: []apiv1.KeyToPath{
+											{
+												Key:  "mao-config",
+												Path: "config",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		if err := createDeployment(testConfig, maoManifest); err != nil {
+			return err
+		}
+
+		err := wait.Poll(pollInterval, timeoutPoolClusterApiDeploymentInterval, func() (bool, error) {
+			if clusterApiDeployment, err := testConfig.KubeClient.AppsV1beta2().Deployments("default").Get("clusterapi-apiserver", metav1.GetOptions{}); err == nil {
```

Comment:
another nice validation would be to verify asking for a machineSet return empty

Reply:
To be done later. Once we flip to the openshift/cluster-api-provider-aws, I would like to create the machine/machinesets as well. So we can actually list the machines as well.

Reply:
Maybe provide `--dry` option to the machine controller, that will just simulate the machine provisioning instead of doing that for real.

Reply:
We just wanna test the api is responsible, `--dry` make sense, we probably don't want to stop the dev of the operator logic here because there's something broken in one actuator for now, also note there will be more than aws.

Reply:
Adddressed